src/components/product/product.service.ts

```diff
@@ -1,140 +1,147 @@
-  ReadProductOutputDto,
-  UpdateProductInput,
-  UpdateProductInputDto,
-  UpdateProductOutputDto,
-} from './product.dto';
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { DatabaseService, PropertyUpdaterService } from '../../core';
 import { generate } from 'shortid';
+import { ISession } from '../auth';
+import { Product, CreateProduct, UpdateProduct } from './dto';
 
 @Injectable()
 export class ProductService {
-  constructor(private readonly db: DatabaseService) {}
-  async create(input: CreateProductInput): Promise<CreateProductOutputDto> {
-    const response = new CreateProductOutputDto();
-    const session = this.db.driver.session();
+  constructor(
+    private readonly db: DatabaseService,
+    private readonly propertyUpdater: PropertyUpdaterService,
+  ) {}
+
+
+  async create(input: CreateProduct, session: ISession): Promise<Product> {
     const id = generate();
-    await session
-      .run(
-        'MERGE (product:Product {active: true, owningOrg: "seedcompany", id: $id}) ON CREATE SET product.id = $id, product.type  = $type, product.books=$books,product.mediums = $mediums,product.purposes=$purposes,product.approach=$approach,product.methodology=$methodology, product.timestamp = datetime() RETURN product.id as id, product.type as type,product.books as books, product.mediums as mediums,product.purposes as purposes,product.approach as approach, product.methodology as methodology',
-        {
-          id,
-          type: input.type,
-          books: input.books,
-          mediums: input.mediums,
-          purposes: input.purposes,
-          approach: input.approach,
-          methodology: input.methodology,
-        },
-      )
-      .then(result => {
-        response.product.id = result.records[0].get('id');
-        response.product.type = result.records[0].get('type');
-        response.product.books = result.records[0].get('books');
-        response.product.mediums = result.records[0].get('mediums');
-        response.product.purposes = result.records[0].get('purposes');
-        response.product.approach = result.records[0].get('approach');
-        response.product.methodology = result.records[0].get('methodology');
-      })
-      .catch(error => {
-        console.log(error);
-      })
-      .then(() => session.close());
-
-    return response;
+    const acls = {
+      canReadType: true,
+      canEditType: true,
+      canReadBooks: true,
+      canEditBooks: true,
+      canReadMediums: true,
+      canEditMediums: true,
+      canReadPurposes: true,
+      canEditPurposes: true,
+      canReadApproach: true,
+      canEditApproach: true,
+      canReadMethodology: true,
+      canEditMethodology: true,
+    };
+
+    try {
+      await this.propertyUpdater.createNode({
+        session,
+        input: { id, ...input },
+        acls,
+        baseNodeLabel: 'Product',
+        aclEditProp: 'canCreateProduct',
+      });
+    } catch (e) {
+      console.log(e);
```

Comment:
We do have a logger service. I've done a poor job communicating it.
It can be injected in the constructor like so,
```tsx
@Logger('colon:grouped:name') private readonly logger: ILogger,
```
`AuthService` is a good example of usage.
`logging.example.yaml` is an example for local configuration.

Reply:
Perfect! Just switched over to logger